go-controller/pkg/ovndbmanager/ovndbmanager.go

```diff
@@ -0,0 +1,217 @@
+	}
+	for _, knownMember := range knownMembers {
+		server := strings.Split(knownMember, ":")
+		if len(server) < 3 {
+			klog.Warningf("Failed to parse known member: %s", knownMember)
+			continue
+		}
+		knownServers = append(knownServers, server[1])
+	}
+	out, stderr, err := appCtl("cluster/status", dbName)
+	if err != nil {
+		klog.Warningf("Unable to get cluster status for: %s, stderr: %v, err: %v", db, stderr, err)
+		return
+	}
+	r, _ := regexp.Compile(`([a-z0-9]{4}) at ((ssl|tcp):\[?[a-z0-9.:]+\]?)`)
+	members := r.FindAllStringSubmatch(out, -1)
+	kickedMembersCount := 0
+	for _, member := range members {
+		if len(member) < 3 {
+			klog.Warningf("Unable to find parse member: %s", member)
+			return
+		}
+		matchedServer := strings.Split(member[2], ":")
+		if len(matchedServer) < 3 {
+			klog.Warningf("Unable to parse address portion of the member entry: %s", matchedServer)
+			return
+		}
+		memberFound := false
+		for _, knownServer := range knownServers {
+			if knownServer == matchedServer[1] {
+				memberFound = true
+				break
+			}
+		}
+		if !memberFound && (len(members)-kickedMembersCount) > 3 {
+			// unknown member and we have enough members its safe to kick the unknown address
+			klog.Infof("Unknown Raft member found: %s, %s...kicking", member[1], member[2])
+			_, stderr, err = appCtl("cluster/kick", dbName, member[1])
+			if err != nil {
+				// warn only: we might fail to kick since other nodes will also be trying to kick the member
+				klog.Warningf("Error while kicking old Raft member: %s, for address: %s in db: %s,"+
+					"stderr: %v, err: %v", member[1], member[2], db, stderr, err)
+				continue
+			}
+			kickedMembersCount = kickedMembersCount + 1
+		}
+	}
+}
+
+// ensureDBHealth ensures that if the db is clustered, it is healthy by calling the cluster-check command
+// if the clustered db is corrupt, the db will be deleted and kube-master pod needs to be started again.
+func ensureDBHealth(db string) {
+	stdout, stderr, err := util.RunOVSDBTool("check-cluster", db)
+	if err != nil {
+		// backup the db by renaming it and then stop the nb/sb ovsdb process.
+		klog.Fatalf("Error occured during checking of clustered db "+
+			"db: %s,stdout: %q, stderr: %q, err: %v",
+			db, stdout, stderr, err)
+		dbFile := filepath.Base(db)
+		backupFile := strings.TrimSuffix(dbFile, filepath.Ext(dbFile)) +
+			time.Now().UTC().Format("2006-01-02_150405") + "db_bak"
+		backupDB := filepath.Join(filepath.Dir(db), backupFile)
+		err := os.Rename(db, backupDB)
+		if err != nil {
+			klog.Warningf("Failed to back up the db to backupFile: %s", backupFile)
+		} else {
+			klog.Infof("Backed up the db to backupFile: %s", backupFile)
+			var dbName string
+			var appCtl func(args ...string) (string, string, error)
+			if strings.Contains(db, "ovnnb") {
+				dbName = "OVN_Northbound"
+				appCtl = util.RunOVNNBAppCtl
+			} else {
+				dbName = "OVN_Southbound"
+				appCtl = util.RunOVNSBAppCtl
+			}
+			_, stderr, err := appCtl("exit")
```

Comment:
if we exit ovsdb-server, dont we need to restart this entire pod? Should this program also exit?

Reply:
it doesn't need to. the nbdb/sbdb containers will restart, but this container can keep running. i validated that it's able to reconnect to the new server instance since the address remains the same.

Reply:
ok cool